framework/auth/core.py

```diff
@@ -1331,6 +1336,29 @@ def n_projects_in_common(self, other_user):
         """Returns number of "shared projects" (projects that both users are contributors for)"""
         return len(self.get_projects_in_common(other_user, primary_keys=True))
 
+    def get_node_comment_timestamps(self, node, page):
```

Comment:
This implementation seems overly confusing to me. I think I understand what it's doing but it seems to be getting there in a non-obvious way. Also, you define default_timestamp and then don't use it inside the page == 'files' block.